src/pages/dashboard.tsx

```diff
@@ -17,35 +19,59 @@ import {
     Text,
     Small,
     Main,
-    LineBreak, UnorderedList, ListItem
+    LineBreak, UnorderedList, ListItem, Show
 } from './../components/SimpleComponents';
-import {Input} from './../components/SmartComponents';
+
+import {Input, LabelValue, Sidebar, SidebarCompanion} from './../components/SmartComponents';
 
 import CONSTANTS from './../config/CONSTANTS';
 
 import {UserService} from './../components/Users';
-import UtilityService, {getBaseUrlWithProtocol} from '../services/UtilityService';
-import TypeService from '../services/TypeService';
+import {GamerRelationshipService} from './../components/GamerRelationships';
+import {GamerSearchInput, GamerAdd, GamerService, GamerLinkList} from '../components/gamer';
+import {Gamer} from '../components/gamer/GamerTypes';
 
 //===----=---=-=--=--===--=-===----=---=-=--=--===--=-===----=---=-=--=--===--=-//
 
 
 let DashboardPage = ({baseUrl}) => {
-
     let router = useRouter();
-    let user = UserService.getUser();
+
+    let [hasMounted, setHasMounted] = useState(false);
+    let [loading, setLoading] = useState(true);
+    let [dataHasLoaded, setDataHasLoaded] = useState(false);
+    let [newUserSearch, setNewUserSearch] = useState('search');
+    let [error, setError] = useState(null);
+
+    let [user, setUser] = useState(null);
+    let [gamerRelationships, setGamerRelationships] = useState([]);
+    let mainGamer = gamerRelationships.length > 0 && gamerRelationships.filter(({type}) => type === 'self')[0] as Gamer;
+    console.log(mainGamer);
+
+    hasMounted === false && setHasMounted(true);
+
+    useEffect(() => {
```

Comment:
This page is absolutely crazy, Zach, I'm sure you'll have some ways to make this better and optimize it, so I'd love your help figuring that out. Login redirect and data structure here gets confusing when it tries to load the mainGamer without having one, or the user before it's been verified that it exists, etc.